internal/ethapi/api_test.go

```diff
@@ -72,117 +73,140 @@ func TestTransaction_RoundTripRpcJSON(t *testing.T) {
 			t.Fatalf("test %d: stx changed, want %x have %x", i, want, have)
 		}
 
-		//  rpcTransaction
+		// rpcTransaction
 		rpcTx := newRPCTransaction(tx, common.Hash{}, 0, 0, 0, nil, config)
 		if data, err := json.Marshal(rpcTx); err != nil {
 			t.Fatalf("test %d: marshalling failed; %v", i, err)
 		} else if err = tx2.UnmarshalJSON(data); err != nil {
 			t.Fatalf("test %d: unmarshal failed: %v", i, err)
 		} else if want, have := tx.Hash(), tx2.Hash(); want != have {
 			t.Fatalf("test %d: tx changed, want %x have %x", i, want, have)
+		} else {
+			want, have := tt.Bs, string(data)
+			require.JSONEqf(t, want, have, "test %d: rpc json not match, want %s have %s", i, want, have)
 		}
 	}
 }
 
-func allTransactionTypes(addr common.Address, config *params.ChainConfig) []types.TxData {
-	return []types.TxData{
-		&types.LegacyTx{
-			Nonce:    5,
-			GasPrice: big.NewInt(6),
-			Gas:      7,
-			To:       &addr,
-			Value:    big.NewInt(8),
-			Data:     []byte{0, 1, 2, 3, 4},
-			V:        big.NewInt(9),
-			R:        big.NewInt(10),
-			S:        big.NewInt(11),
-		},
-		&types.LegacyTx{
-			Nonce:    5,
-			GasPrice: big.NewInt(6),
-			Gas:      7,
-			To:       nil,
-			Value:    big.NewInt(8),
-			Data:     []byte{0, 1, 2, 3, 4},
-			V:        big.NewInt(32),
-			R:        big.NewInt(10),
-			S:        big.NewInt(11),
+type txData struct {
+	Tx types.TxData
+	Bs string
+}
+
+func allTransactionTypes(addr common.Address, config *params.ChainConfig) []txData {
+	return []txData{
+		{
+			Tx: &types.LegacyTx{
+				Nonce:    5,
+				GasPrice: big.NewInt(6),
+				Gas:      7,
+				To:       &addr,
+				Value:    big.NewInt(8),
+				Data:     []byte{0, 1, 2, 3, 4},
+				V:        big.NewInt(9),
+				R:        big.NewInt(10),
+				S:        big.NewInt(11),
+			},
+			Bs: `{"blockHash":null,"blockNumber":null,"from":"0x71562b71999873db5b286df957af199ec94617f7","gas":"0x7","gasPrice":"0x6","hash":"0x5f3240454cd09a5d8b1c5d651eefae7a339262875bcd2d0e6676f3d989967008","input":"0x0001020304","nonce":"0x5","to":"0xdead000000000000000000000000000000000000","transactionIndex":null,"value":"0x8","type":"0x0","chainId":"0x539","v":"0xa96","r":"0xbc85e96592b95f7160825d837abb407f009df9ebe8f1b9158a4b8dd093377f75","s":"0x1b55ea3af5574c536967b039ba6999ef6c89cf22fc04bcb296e0e8b0b9b576f5"}`,
```

Comment:
Nit: it's not clear to me what Bs is short for. Please use `Want`.

Reply:
Sorry, just want to indent with `Tx` :)